requirements.txt

```diff
@@ -1,9 +1,11 @@
+backports.cached-property==1.0.0.post2
```

Comment:
I used `@cached_property` just because some of it felt kind of silly (like stuff in templates where you do `if this then render this`).  Probably not a huge deal, if we don't want this dep I can just use plain properties everywhere.  I think cached_property is part of functools for python > 3.8?

Reply:
I have a separate PR up to bump the minimum required Python version to 3.9, at which point you're right we'll be able to just pull this from `functools` and not need the backport. Maybe for this PR we remove this dep, switch to plain properties, and once 3.9 lands we update.

Reply:
Update: I merged in the 3.9 requirement, so if you rebase/merge latest `main` you'll be able to import this from functools 🎉

Reply:
Rebased and updated import, thanks.